### PR TITLE
Bravin format

### PR DESCRIPTION
**Pull request by bravint**

### CODE REVIEW

File: src/client/App.jsx [hunk 1]
```diff
@@ -33,7 +33,7 @@ const App = () => {
         const fetchedUser = async () => {
             try {
                 const response = await fetch(`${USER_URL.USER_ROOT}${id}`, {
-                    method: 'GET',
+                    method: `${FETCH_METHOD.GET}`,
```
Comment (alexamaingard) [inline src/client/App.jsx:36]: Nice change

**Comment by bravint** [comment]
most of these changes are prettier doing its thing,

code changes are in src/client/pages/leaderboard

**Review by polarbear23 — APPROVED** [review]

File: src/client/pages/forum/forms/PostForm.jsx [hunk 2]
```diff
@@ -1,88 +1,99 @@
...
-			name === "Cosplay" ||
-			name === "Manga" ||
-			name === "Tv/Film"
-		) {
-			setCategories([...categories, name]);
+    const { posts, setPosts } = props;
+
+	const initialPost = {
+		title: '',
+        content: '',
```
Comment (alexamaingard) [inline src/client/pages/forum/forms/PostForm.jsx:13]: Fix tabbing please

File: src/client/pages/forum/forms/PostForm.jsx [hunk 3]
```diff
@@ -1,88 +1,99 @@
...
+            name === 'Tv/Film'
+        ) {
+            setCategories([...categories, name]);
+        }
+    };
+
+    const submitNewPostHandler = (event) => {
+        event.preventDefault();
+
+		const newPost = {
```
Comment (alexamaingard) [inline src/client/pages/forum/forms/PostForm.jsx:40]: Fix tabbing please

**Review by hannah68 — APPROVED** [review]

**Review by alexamaingard — CHANGES_REQUESTED** [review]

**Review by alexamaingard — APPROVED** [review]

File: src/client/pages/forum/forms/PostForm.jsx (hunk 2 above)
Comment (bravint) [reply src/client/pages/forum/forms/PostForm.jsx:13]: done

File: src/client/pages/forum/forms/PostForm.jsx (hunk 3 above)
Comment (bravint) [reply src/client/pages/forum/forms/PostForm.jsx:40]: done

File: src/client/App.jsx (hunk 1 above)
Comment (bravint) [reply src/client/App.jsx:36]: thank you for your opinion

Comment (vherus) [inline src/client/App.jsx:36]: I like the approach here but try not to confuse `fetch` the function with a HTTP Request: `HTTP_METHOD.GET` may be a better fit

File: src/client/App.jsx [hunk 4]
```diff
@@ -62,10 +62,20 @@ const App = () => {
             />
             <LeftMenu />
             <Routes>
-                <Route path={INT_LINK.HOME} element={<Home />} />
+                <Route 
+                    path={INT_LINK.HOME}
```
Comment (vherus) [inline src/client/App.jsx:66]: What does INT mean here? I'm hazarding a guess that it's INTERNAL, if it is I'd recommend being more explicit. Think of the poor junior dev who's going to maintain this in 5 years ;)

File: src/client/Footer.jsx [hunk 5]
```diff
@@ -1,46 +1,47 @@
...
-          <li className="footer__lists--item">
-            <Link to="/" className="disabled-link">
-              Terms and rules
-            </Link>
-          </li>
-          <li className="footer__lists--item">
-            <Link to="/contact">Contact Us</Link>
-          </li>
-        </ul>
+    return (
```
Comment (vherus) [inline src/client/Footer.jsx:7]: Nice. Tabs are indeed superior to spaces!

File: src/client/LeftMenu.jsx [hunk 6]
```diff
@@ -1,52 +1,54 @@
...
-  }
-  const changeToLight = () => {
-    setDarkMode(false);
-  }
+    const changeToDark = () => {
+        setDarkMode(true);
+    };
+    const changeToLight = () => {
+        setDarkMode(false);
+    };
```
Comment (vherus) [inline src/client/LeftMenu.jsx:13]: You could turn this into a single switch by changing `darkMode` to the opposite of what it currently is:

```js
const switchDarkMode = () => setDarkMode(!darkMode);
```

File: src/client/pages/forum/forms/PostForm.jsx [hunk 7]
```diff
@@ -1,88 +1,99 @@
...
+
+        setNewPost({ ...newPost, [name]: value });
+
+        if (
+            name === 'Gaming' ||
+            name === 'Fan-Fiction' ||
+            name === 'Cosplay' ||
+            name === 'Manga' ||
+            name === 'Tv/Film'
+        ) {
```
Comment (vherus) [inline src/client/pages/forum/forms/PostForm.jsx:32]: I can feel the pain inside you for this. Take a look at `Array.prototype.includes` for an idea about how to make this less gross: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes

**Review by vherus — APPROVED** [review]
It seems like you've swung the other direction and put the majority of your work into a single commit, I don't think that's what Ed was trying to outline. The amount of changes involved is the real problem here.

All of the formatting you did should've been on its own branch, in its own PR. The Post functionality should be in its own PR. The leaderboard changes should be in its own PR.

Try to consider whether what you're about to work on should be lumped in with your current feature - if it can stand on its own, it should be on a different branch for a different PR.

**Pull request merged**